### PR TITLE
E2E: revise the WordPress.com Free signup spec to the new, simplified signup flow.

### PR DESCRIPTION
**Pull request by worldomonation**
Related to paYE8P-2bD-p2

## Proposed Changes

This PR repurposes the existing `Plans: Free` flow to to the newly simplified `Free Signup` flow.

This new flow is sufficiently different from existing ones that new coverage was deemed necessary in p1676573321667029-slack-CQD1HH4MA.

Key changes:
- addition of `FreeSignupFlow` POM, similar to the existing `StartSiteFlow`.
- added coverage of the `Free Signup` flow available at `setup/free/freeSetup`.

## Testing Instructions

Ensure the following build configurations are passing:
  - [ ] Pre-Release E2E

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: packages/calypso-e2e/src/lib/flows/free-signup-flow.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	 *
+	 * Note that in this flow, the site name will form part of the domain name.
+	 *
+	 * Example:
+	 * 	freetestsite1234 -> freetestsite.wordpress.com
+	 *
+	 * @param {string} name Name of the site.
+	 */
+	async enterSitename( name: string ): Promise< void > {
+		await this.page.getByRole( 'textbox', { name: 'My WebSite' } ).fill( name );
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/flows/free-signup-flow.ts:41]: This method and the one below it `enterDescription` both do _very_ similar things, but I decided it was better to have named functions for ease of understanding.

If we want to condense these into one method, an alternative approach is to have a `enterSiteDetails` method that takes a required parameter `name` and optional param `description`.

File: packages/calypso-e2e/src/lib/flows/free-signup-flow.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	async pickDesign( name: string ): Promise< void > {
+		await this.page.getByText( name, { exact: true } ).click();
+
+		// FSE styles present a screen to choose a style and a button to
+		// continue with default options, while normal themes only show
+		// a preview and a button to select the theme.
+		await Promise.race( [
+			this.clickButton( 'Continue' ),
+			this.clickButton( `Start with ${ name }` ),
+		] );
```
Comment (worldomonation) [inline packages/calypso-e2e/src/lib/flows/free-signup-flow.ts:67]: I'm intentionally not dealing with the FSE styles here for simplicity's sake.

File: test/e2e/specs/onboarding/signup__free.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	RestAPIClient,
+	UserSignupPage,
+} from '@automattic/calypso-e2e';
+import { Page, Browser } from 'playwright';
+import { apiCloseAccount } from '../shared';
+import type { NewUserResponse } from '@automattic/calypso-e2e';
+
+declare const browser: Browser;
+
+describe( 'Signup: Create a WordPress.com Free site as a new user', function () {
```
Comment (worldomonation) [inline test/e2e/specs/onboarding/signup__free.ts:18]: Notice that I have discontinued use of the `DataHelper.createSuiteTitle` helper method.

That was a leftover from the Selenium days when we used to add various metadata to the suite title, such as viewport name, whether the test was to run in parallel, etc. 

We have moved far away from those days and so I think it's justified to no longer use the helper method.

File: packages/calypso-e2e/src/lib/flows/free-signup-flow.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	 * Enters the site name.
+	 *
+	 * Note that in this flow, the site name will form part of the domain name.
+	 *
+	 * Example:
+	 * 	freetestsite1234 -> freetestsite.wordpress.com
+	 *
+	 * @param {string} name Name of the site.
+	 */
+	async enterSitename( name: string ): Promise< void > {
```
Comment (karenroldan) [inline packages/calypso-e2e/src/lib/flows/free-signup-flow.ts]: **nitpick (non-blocking):** `enterSitename` to `enterSiteName`

File: packages/calypso-e2e/src/lib/flows/free-signup-flow.ts (hunk 1 above)
Comment (karenroldan) [reply packages/calypso-e2e/src/lib/flows/free-signup-flow.ts:41]: **thought:** I don't have a strong preference for separating the methods or condense into one method. The current approach is fine!

File: test/e2e/specs/onboarding/signup__free.ts (hunk 3 above)
Comment (karenroldan) [reply test/e2e/specs/onboarding/signup__free.ts:18]: > That was a leftover from the Selenium days when we used to add various metadata to the suite title

This is good to know!

**Review by karenroldan — APPROVED** [review]
The changes look good to me! Thank you for working on this 🚀

**Pull request merged**